backend/public-graph/graph/resolver.go

```diff
@@ -2963,9 +2964,9 @@ func (r *Resolver) ProcessPayload(ctx context.Context, sessionSecureID string, e
 	updateSpan, _ := tracer.StartSpanFromContext(ctx, "public-graph.pushPayload", tracer.ResourceName("doSessionFieldsUpdate"))
 	defer updateSpan.Finish()
 
-	excluded := r.isSessionUserExcluded(ctx, sessionObj)
+	excluded, reason := r.isSessionExcluded(ctx, sessionObj, sessionHasErrors)
```

Comment:
`sessionHasErrors` is only looking at the current batch of errors. 
```sessionHasErrors := len(errors) > 0```
with the logic below
```
        doUpdate := ... ||
		(sessionHasErrors && (sessionObj.HasErrors == nil || !*sessionObj.HasErrors))
	if doUpdate && !excluded {
		fieldsToUpdate := model.Session{
			PayloadUpdatedAt: &now, BeaconTime: beaconTime, HasUnloaded: hasSessionUnloaded, Processed: &model.F, ObjectStorageEnabled: &model.F, DirectDownloadEnabled: false, Chunked: &model.F, Excluded: false,
		}
```
that means that a session will not be updated by `PushPayload` until it gets an error (when the `FilterSessionsWithoutError` setting is enabled), right? i don't think that's right, as subsequent push payloads may not have an error.

ie, consider a project with `FilterSessionsWithoutError=true`
1. PushPayload with no errors -> does not update the session
2. PushPayload with an error -> updates the session
3. PushPayload with no errors -> does not update the session
4. PushPayload with no errors -> does not update the session
`PushPayloads` 3 and 4 will not update the payload time, so the session might be missing events in processing.

Reply:
> sessionHasErrors is only looking at the current batch of errors.
sessionHasErrors := len(errors) > 0

Are you considering this block of code as well:

https://github.com/highlight/highlight/blob/cdc89e81676d125389a73cf0d5b17296ea9a6108/backend/public-graph/graph/resolver.go#L2954-L2961

Reply:
We walked through the code together and the current implementation looks to be good.